yt_dlp/extractor/radiko.py

```diff
@@ -112,22 +112,26 @@ def _extract_formats(self, video_id, station, is_onair, ft, cursor, auth_token,
 
             time_to_skip = None if is_onair else cursor - ft
 
+            domain = urllib.parse.urlparse(playlist_url).netloc
             subformats = self._extract_m3u8_formats(
                 playlist_url, video_id, ext='m4a',
-                live=True, fatal=False, m3u8_id=None,
+                live=True, fatal=False, m3u8_id=domain,
+                note=f'Downloading m3u8 information from {domain}',
                 headers={
                     'X-Radiko-AreaId': area_id,
                     'X-Radiko-AuthToken': auth_token,
                 })
             for sf in subformats:
-                domain = sf['format_id'] = compat_urllib_parse.urlparse(sf['url']).netloc
-                if re.match(r'^[cf]-radiko\.smartstream\.ne\.jp$', domain):
+                if re.fullmatch(r'[cf]-radiko\.smartstream\.ne\.jp', domain):
                     # Prioritize live radio vs playback based on extractor
                     sf['preference'] = 100 if is_onair else -100
                 if not is_onair and url_attrib['timefree'] == '1' and time_to_skip:
                     sf['downloader_options'] = {'ffmpeg_args': ['-ss', time_to_skip]}
             formats.extend(subformats)
 
+        if not formats:
+            self.raise_no_formats('No formats found! You should force re-authenticate using --extractor-args radiko:force_reauth or clear cache.')
```

Comment:
why can't we just automatically re-authenticate + re-extract here? That's what other similar extractors do

Reply:
Premium-only archives (without paying) falls here too

Reply:
Oh yes, I haven't implemented auth

makes sense

Reply:
Done